pkgs/development/libraries/glfw/3.x-wayland-minecraft.nix

```diff
@@ -1,64 +1,8 @@
-{ stdenv, lib, fetchFromGitHub, fetchpatch2, cmake, extra-cmake-modules
-, libGL, wayland, wayland-protocols, libxkbcommon, libdecor
-}:
-
-stdenv.mkDerivation {
-  version = "unstable-2023-06-01";
+{ glfw }:
+glfw.overrideAttrs (old: {
```

Comment:
This is generally not a recommended pattern as a user could have something like the following overlay which would cause an infinite recursion:

```nix
final: prev: {
  glfw = final.glfw-wayland-minecraft
}
```

What you could do instead is have some kind of `common.nix` that you can import and use in both derivations.

Reply:
> What you could do instead is have some kind of `common.nix` that you can import and use in both derivations.

So do you imply that we should to keep glfw-wayland-minecraft in parallel with the original glfw, rather than making it a package option?

Reply:
Hmm, I guess making it a package option of the original glfw is also a good way to do this!